modules/apps/account/account-service/src/main/java/com/liferay/account/service/impl/AccountEntryLocalServiceImpl.java

```diff
@@ -271,6 +280,30 @@ public AccountEntry addOrUpdateAccountEntry(
 		return accountEntryPersistence.update(accountEntry);
 	}
 
+	@Override
+	public AccountEntry addOrUpdateContact(
+			AccountEntry accountEntry, String facebookSn, String jabberSn,
+			String skypeSn, String smsSn, String twitterSn)
+		throws PortalException {
+
+		Contact contact = accountEntry.fetchContact();
```

Comment:
plus if the get is throwing the not found exception you won't add the contact

Reply:
it should throw the exception only if I don't hae permission to view the account and in this case I think it's ok that I'm not able to create the contact